src/lib/router.ts

```diff
@@ -112,10 +109,61 @@ async function navigate(
     return;
   }
 
+  setTitle(loadedRoute);
   activeRouteStore.set(loadedRoute);
   isLoading.set(false);
 }
 
+function setTitle(loadedRoute: LoadedRoute) {
+  const title: string[] = [];
+
+  if (loadedRoute.resource === "home") {
+    title.push("Pinned projects");
+  } else if (loadedRoute.resource === "seeds") {
+    title.push(loadedRoute.params.baseUrl.hostname);
+  } else if (loadedRoute.resource === "booting") {
+    // Don't set the title while booting to avoid flickering.
+  } else if (loadedRoute.resource === "loadError") {
+    title.push("Load error");
+    title.push("Radicle");
+  } else if (loadedRoute.resource === "notFound") {
+    title.push("Page not found");
+    title.push("Radicle");
+  } else if (loadedRoute.resource === "session") {
+    title.push("Authenticating");
+    title.push("Radicle");
+  } else if (loadedRoute.resource === "projects") {
+    if (loadedRoute.params.view.resource === "tree") {
+      title.push(loadedRoute.params.project.name);
+      title.push(loadedRoute.params.project.description);
+    } else if (loadedRoute.params.view.resource === "commits") {
+      title.push(loadedRoute.params.view.commit.commit.summary);
+      title.push("commit");
+    } else if (loadedRoute.params.view.resource === "history") {
+      title.push(loadedRoute.params.project.name);
+      title.push("history");
+    } else if (loadedRoute.params.view.resource === "issue") {
+      title.push(loadedRoute.params.view.params.loadedIssue.title);
+      title.push("issue");
+    } else if (loadedRoute.params.view.resource === "issues") {
+      title.push(loadedRoute.params.project.name);
+      title.push("issues");
+    } else if (loadedRoute.params.view.resource === "patches") {
+      title.push(loadedRoute.params.project.name);
+      title.push("patches");
+    } else if (loadedRoute.params.view.resource === "patch") {
+      title.push(loadedRoute.params.view.params.loadedPatch.title);
+      title.push("patch");
+    } else {
+      utils.unreachable(loadedRoute.params.view);
+    }
```

Comment:
To keep things modular I would move this code into a function in `app/views/project/router`. This way we keep the size of this code in check and we only have to change one file if we want to adjust the project routing.

Reply:
3b1495657fa10d51157d6d2bcc76d8709490fee9